src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -37,6 +37,10 @@
     /** A pattern matching the symbol: "/". */
     private static final Pattern SLASH_PATTERN = Pattern.compile("[\\\\/]");
 
+    /** A pattern to find the string: "// violation". */
+    private static final Pattern VIOLATION_PATTERN = Pattern
+            .compile(".*//\\s*violation(?:$|\\W+.*$)");
```

Comment:
why we need `?:`

Reply:
because we don't need to capture the match, right?